src/librustc_trans/trans/mir/did.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+//! Code for translating references to other items (DefIds).
+
+use rustc::front::map;
+use rustc::middle::ty::{self, Ty, TypeVariants};
+use rustc::middle::subst::Substs;
+use rustc::middle::const_eval;
+use rustc::middle::def_id::DefId;
+use trans::common::Block;
+use trans::base;
+use trans::expr;
+use trans::monomorphize;
+use trans::inline;
+
+use super::MirContext;
+use super::operand::{OperandRef, OperandValue};
+
+impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
+    /// Translate reference to item.
+    pub fn trans_item_ref(&mut self,
+                          bcx: Block<'bcx, 'tcx>,
+                          ty: Ty<'tcx>,
+                          substs: &'tcx Substs<'tcx>,
+                          did: DefId)
+                          -> OperandRef<'tcx> {
+        match ty.sty {
```

Comment:
Okay, I’m convinced now this is not only a hack but is also wrong (e.g. it will panic compiling a function that’s defined in `impl` block).

However, I won’t be able to get to it for at least a week and a half. I’ll go ahead and mark the issue as [WIP].